l10n_nl_postcodeapi/models/res_partner.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright (C) 2013-2015 Therp BV (<http://therp.nl>).
+#
+#    @autors: Stefan Rijnhart, Ronald Portier
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from lxml import etree
+from openerp import models, api, exceptions, _
+from openerp.tools import ormcache
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @ormcache(skiparg=2)
+    def get_provider_obj(self):
+        self._cr.execute("""
+                SELECT value FROM ir_config_parameter
+                WHERE key = %s""", ('l10n_nl_postcodeapi.apikey',))
+        row = self._cr.fetchone()
+        if row and row[0] != 'Your API key' and row[0].strip():
+            from pyPostcode import Api
+            provider = Api(row[0].strip())
+            test = provider.getaddress('1053NJ', '334T')
+            if not test or not test._data:
```

Comment:
then it errors out later. But this way, every `on_change` triggers two requests, that's not desireable, is it?